src/isar/robot/robot.py

```diff
@@ -0,0 +1,261 @@
+        if self.robot_status_thread:
+            self.robot_status_thread.wait_for_thread()
+        if self.robot_task_status_thread:
+            self.robot_task_status_thread.wait_for_thread()
+        if self.start_mission_thread:
+            self.start_mission_thread.wait_for_thread()
+        self.robot_status_thread = None
+        self.robot_task_status_thread = None
+        self.start_mission_thread = None
+
+    def run_start_mission_thread(self, mission_or_task: Mission | Task):
+        retries = 0
+        started_mission = False
+        try:
+            while not started_mission:
+                try:
+                    if settings.RUN_MISSION_BY_TASK:
+                        self.robot.initiate_task(
+                            mission_or_task
+                        )
+                    else:
+                        self.robot.initiate_mission(mission_or_task)
+                except RobotException as e:
+                    retries += 1
+                    self.logger.warning(
+                        f"Initiating failed #: {str(retries)} "
+                        f"because: {e.error_description}"
+                    )
+
+                    if retries >= settings.INITIATE_FAILURE_COUNTER_LIMIT:
+                        error_description = (
+                            f"Mission will be cancelled after failing to initiate "
+                            f"{settings.INITIATE_FAILURE_COUNTER_LIMIT} times because: "
+                            f"{e.error_description}"
+                        )
+
+                        self._trigger_event(
+                            self.queues.robot_mission_failed,
+                            ErrorMessage(
+                                error_reason=e.error_reason,
+                                error_description=error_description,
+                            ),
+                        )
+                started_mission = True
+        except RobotInfeasibleTaskException as e:
+            error_description = (
+                f"Failed to initiate task "
+                f"{str(mission_or_task.id)[:8]} after retrying "
+                f"{retries} times because: "
+                f"{e.error_description}"
+            )
+            self._trigger_event(
+                self.queues.robot_task_status_failed,
+                ErrorMessage(
+                    error_reason=e.error_reason, error_description=error_description
+                ),
+            )
+        except RobotInfeasibleMissionException as e:
+            self._trigger_event(
+                self.queues.robot_mission_failed,
+                ErrorMessage(
+                    error_reason=e.error_reason, error_description=e.error_description
+                ),
+            )
+        self.robot_task_status_thread = ThreadedRequest(
+            request_func=self.run_robot_task_status_thread
+        )
+        self.robot_task_status_thread.start_thread(name="Robot task status thread")
```

Comment:
how is this stopped when there is no mission active?

Reply:
The thread exits when it transitions out of monitor, and it is then cleaned up when receiving another mission

Reply:
I will change it so that the thread stays alive, but only sleeps while no mission or task is active. This way we avoid the cost of having to start another thread when a new task starts. Alternatively I might make it start when we detect that we enter monitor, then exit when we leave monitor.